cmd/declaration_extraction/declaration_extraction_test.go

```diff
@@ -0,0 +1,125 @@
+package declaration_extraction
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+func TestExtractGlbl(t *testing.T) {
+	file, err := os.Open("./test.cx")
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	got := extractGlbl(file, "test.cx")
+
+	want := []Declaration{
+		Declaration{
+			PackageID:   "hello",
+			FileID:      "test.cx",
+			StartOffset: 0,
+			Length:      3,
+			Name:        "apple",
+		},
+		Declaration{
+			PackageID:   "hello",
+			FileID:      "test.cx",
+			StartOffset: 7,
+			Length:      3,
+			Name:        "banana",
+		},
+	}
+
+	//Check if any declaration were detected
+	if got == nil {
+		t.Error("No Global Declarations")
+	}
+
+	for i := range got {
+		if got[i] != want[i] {
+			t.Errorf("got %+v   want %+v\n", got[i], want[i])
+		}
+	}
+
+}
+
+func TestExtractStrct(t *testing.T) {
+	file, err := os.Open("./test.cx")
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	got := extractStrct(file, "test.cx")
+
+	want := []Declaration{
+		Declaration{
+			PackageID:   "hello",
+			FileID:      "test.cx",
+			StartOffset: 0,
+			Length:      3,
```

Comment:
Can you pls double-check, I don't think these values are the right ones. for `var apple string` the length should be 16. So when we have an offset, we can get the whole declaration by using `sourceCodeBytes[offset:offset+length]`